<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QtWidgetsClass</class>
 <widget class="QMainWindow" name="QtWidgetsClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1121</width>
    <height>825</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtWidgetsApplication1</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox_RGB">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>120</y>
      <width>221</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>RGB</string>
    </property>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>30</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>R</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>30</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>G</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>30</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>B</string>
     </property>
    </widget>
    <widget class="QSlider" name="RSlider">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>20</y>
       <width>171</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="GSlider">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>50</y>
       <width>171</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="BSlider">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>171</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label_Image">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>881</width>
      <height>601</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_ROI">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>420</y>
      <width>221</width>
      <height>141</height>
     </rect>
    </property>
    <property name="title">
     <string>ROI</string>
    </property>
    <widget class="QLineEdit" name="lineEdit_X">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>60</y>
       <width>71</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_Y">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>90</y>
       <width>71</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_X">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>左上座標:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_Y">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>81</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>右下座標:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="roiBtn">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>10</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>ROI</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="ROIBtnChoose">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>10</y>
       <width>71</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>選擇角點</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>230</y>
      <width>221</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>二值化</string>
    </property>
    <widget class="QSlider" name="thresholdSlider">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>191</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="pageStep">
      <number>10</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="sliderPosition">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>310</y>
      <width>221</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>透視變換</string>
    </property>
    <widget class="QPushButton" name="perBtn">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>10</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>透視變換</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_LT">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>40</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_RT">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>40</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="text">
      <string>LeftTop</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="text">
      <string>LeftBtm</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>70</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="text">
      <string>RighBtm</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>40</y>
       <width>41</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="text">
      <string>RighTop</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="perBtnChoose">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>10</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>選擇角點</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_RB">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>70</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_LB">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>70</y>
       <width>51</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>0</y>
      <width>221</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>翻轉圖片</string>
    </property>
    <widget class="QSlider" name="rotate_x">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>20</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-180</number>
     </property>
     <property name="maximum">
      <number>180</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="rotate_y">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>50</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-180</number>
     </property>
     <property name="maximum">
      <number>180</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="rotate_z">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>80</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>-180</number>
     </property>
     <property name="maximum">
      <number>180</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>30</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>x</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>30</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>z</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>30</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>y</string>
     </property>
    </widget>
   </widget>
   <widget class="QTableWidget" name="consoleTable">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>600</y>
      <width>1121</width>
      <height>171</height>
     </rect>
    </property>
    <property name="showGrid">
     <bool>false</bool>
    </property>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>210</number>
    </attribute>
    <column>
     <property name="text">
      <string>Time</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Operation</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Operation Object</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Details</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Note</string>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="cameraBtn">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>570</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>相機</string>
    </property>
   </widget>
   <widget class="QPushButton" name="faceBtn">
    <property name="geometry">
     <rect>
      <x>1010</x>
      <y>570</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>口罩</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1121</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>檔案</string>
    </property>
    <addaction name="openBtn"/>
    <addaction name="saveBtn"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>直方圖</string>
    </property>
    <addaction name="histogramBtn"/>
    <addaction name="equalHistBtn"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>色彩空間轉換</string>
    </property>
    <addaction name="grayBtn"/>
    <addaction name="hsvBtn"/>
   </widget>
   <widget class="QMenu" name="menuai6cj6">
    <property name="title">
     <string>模糊、銳化</string>
    </property>
    <addaction name="blurBtn"/>
    <addaction name="gaussblurBtn"/>
    <addaction name="mediblurBtn"/>
    <addaction name="sharpenBtn"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
   <addaction name="menuai6cj6"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="openBtn">
   <property name="text">
    <string>開啟檔案</string>
   </property>
  </action>
  <action name="saveBtn">
   <property name="text">
    <string>儲存檔案</string>
   </property>
  </action>
  <action name="histogramBtn">
   <property name="text">
    <string>顯示直方圖</string>
   </property>
  </action>
  <action name="equalHistBtn">
   <property name="text">
    <string>直方圖等化</string>
   </property>
  </action>
  <action name="grayBtn">
   <property name="text">
    <string>Gray</string>
   </property>
  </action>
  <action name="hsvBtn">
   <property name="text">
    <string>HSV</string>
   </property>
  </action>
  <action name="gaus_lowBtn">
   <property name="text">
    <string>lower</string>
   </property>
  </action>
  <action name="fourierBtn">
   <property name="text">
    <string>Fourier</string>
   </property>
  </action>
  <action name="gaus_meanBtn">
   <property name="text">
    <string>均值模糊</string>
   </property>
  </action>
  <action name="blurBtn">
   <property name="text">
    <string>均值模糊</string>
   </property>
  </action>
  <action name="gaussblurBtn">
   <property name="text">
    <string>高斯模糊</string>
   </property>
  </action>
  <action name="mediblurBtn">
   <property name="text">
    <string>中值模糊</string>
   </property>
  </action>
  <action name="twoblurBtn">
   <property name="text">
    <string>双边模糊</string>
   </property>
  </action>
  <action name="sharpenBtn">
   <property name="text">
    <string>銳化</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="QtWidgets.qrc"/>
 </resources>
 <connections/>
 <slots>
  <slot>openFile()</slot>
  <slot>ROI()</slot>
  <slot>histogram()</slot>
  <slot>Qthreshold()</slot>
  <slot>QequalizeHist()</slot>
  <slot>change2Gray()</slot>
  <slot>change2HSV()</slot>
  <slot>savefile()</slot>
  <slot>rotate()</slot>
  <slot>QRGB()</slot>
  <slot>perspective()</slot>
 </slots>
</ui>
